### DISCUSSION TITLE
github personal access token

### DISCUSSION THREAD

**Discussion by shmish111** [fluxcd/flux2#1662] [Q&A]
The documentation for bootstrapping flux with github clearly highlights:
> Removing the personal access token will also remove the deploy key.
After I have bootstrapped flux, I really don't want it associated with my personal github account in any way, what happens when I leave my job for example? Once I have bootstrapped, can I change the ssh key (I could generate a new one, just for flux)

**Comment by stefanprodan** [comment]
You can generate a SSH key pair and tell flux to use it during bootstrap: `flux bootstrap github --private-key-file=path/to/id_rsa`

**Comment by squaremo** [comment]
> Once I have bootstrapped, can I change the ssh key (I could generate a new one, just for flux)

Yes, you can replace the secret with a new one, or start with your own secret.

**Replace the secret**

`flux create secret git` will create a new key and create or update the secret you name, then print the public key. You can overwrite the key made by bootstrap, or make a new one and point the GitRepository at it. In either case you will need to add the public key as a deploy key to your GitHub repo.

**Start with your own secret**

Create a key with e.g., `ssh-keygen -f identity`, then (as Stefan suggests) use `flux bootstrap github --private-key=./identity ...` to use it when bootstrapping.
